include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -748,7 +748,7 @@ namespace internal
         // source:
         // https://stackoverflow.com/questions/47981/how-do-you-set-clear-and-toggle-a-single-bit
         // "Changing the nth bit to x"
-        number ^= (-x ^ number) & (1UL << n);
+        number ^= (-static_cast<char>(x) ^ number) & (1UL << n);
```

Comment:
I find that exceedingly difficult to read but probably correct. Among the questions I have are:
* Does this make the assumption that `char` is `signed`? The standard doesn't actually guarantee that, but I think that's what you assume here.
* You don't check that `n` is between zero and 7. That would probably be worth an assertion. 
* Since you're only working on one byte, you don't need `1UL`. You could probably get away with `signed int` or even `signed short`.